API Calls and more

saiflakhani opened the pull request:
1. Added spotify token and sign in values to local storage, and retrieval
2. Added a search form on the ```popup-signed-in``` to allow for quick search
3. Added an API call to the spotify server asynchronously, and added a message callback on the frontend
4. Parsed JSON into a table format. Beautification need
5. Added some documentation
6. Formatting of various JS and HTML files

schezfaz commented on popup-signed-in-script.js:
```diff
@@ -2,4 +2,27 @@ document.querySelector('#sign-out').addEventListener('click', function () {
+});
+
+var resultsPlaceholder = document.getElementById('results');
+var templateSource = document.getElementById('results-template').innerHTML;
```
Hi Saif, are we populating anything here? I don't see the `templateSource` variable being referenced anywhere after assignment

saiflakhani replied:
Yep you're right. Line 22 is dead code. I initially wanted to use the handlebars library instead of writing the entire table <tr> and <td>, but chrome rejected that library from loading. My bad, please remove that.

saiflakhani replied:
I also wanted to say that the search API call is simply boilerplate code. I'm guessing you need some kind of web scraper to monitor the webpage and lookup if it finds a song. 

- I explored the youtube option, but that would require us to integrate Google APIs to get information of the video like title and artist from the video URL.   URL retrieval is present in background.js
- An alternative to that is using a scraper, but it seems youtube has some infrastructure preventing us from doing it. Source: [Stackoverflow](https://stackoverflow.com/questions/1216029/get-title-from-youtube-videos)

schezfaz replied:
> Yep you're right. Line 22 is dead code. I initially wanted to use the handlebars library instead of writing the entire table tr and td, but chrome rejected that library from loading. My bad, please remove that, as well as lines 50-54 of the HTML file of this script.

Haha no worries! Could you commit the required changes to your branch, i'll merge the  request post that!

schezfaz replied:
> I also wanted to say that the search API call is simply boilerplate code. I'm guessing you need some kind of web scraper to monitor the webpage and lookup if it finds a song.
> 
> * I explored the youtube option, but that would require us to integrate Google APIs to get information of the video like title and artist from the video URL.   URL retrieval is present in background.js
> * An alternative to that is using a scraper, but it seems youtube has some infrastructure preventing us from doing it. Source: [Stackoverflow](https://stackoverflow.com/questions/1216029/get-title-from-youtube-videos)

Yup,  was thinking along similar lines! however, we can just extract the title from the browser tab whenever youtube is open instead of scraping the DOM, yeah?

saiflakhani commented:
1. Aligned theme with Spotify, and add search bar animations (Trust me, they're impressive)
2. Remove unnecessary code referenced in the previous commit
3. Moved css to dedicated stylesheet folder
4. Added Font: Gotham, as used by Spotify

schezfaz commented:
Love what you've done with it! Thanks Saif 🥳
